File: src/main/java/com/lannstark/lec15/Lec15Main.java
package com.lannstark.lec15;

import com.kimjingyu.lec15.ListWithJava;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Lec15Main {

  public static void main(String[] args) {
    mapTest();

    ListWithJava listWithJava = new ListWithJava();
    List<Integer> unmodifiableList = listWithJava.getUnmodifiableList();
    unmodifiableList.add(300);

    for (Integer integer : unmodifiableList) {
      System.out.println("integer = " + integer);
    }
  }

  private static void mapTest() {
    // JDK 8까지
    Map<Integer, String> map = new HashMap<>();
    map.put(1, "MONDAY");
    map.put(2, "TUESDAY");

    // JDK 9부터
    Map.of(1, "MONDAY", 2, "TUESDAY");

    for (Integer key : map.keySet()) {
      System.out.println(key + " " + map.get(key));
    }

    for (Map.Entry<Integer, String> entry : map.entrySet()) {
      System.out.println(entry.getKey() + " " + entry.getValue());
    }
  }

  private static void arrayTest() {
    int[] arr = {100, 200};

    for (int i = 0; i < arr.length; i++) {
      System.out.println(i + " " + arr[i]);
    }
  }

}

File: src/main/kotlin/com/kimjingyu/lec15/Lec15Main.kt
package com.kimjingyu.lec15

import java.util.Collections

class ListWithJava() {
    val list = listOf(100, 200)
    val unmodifiableList = Collections.unmodifiableList(list)
    val modifiableList = mutableListOf(100, 200)
}

fun main() {

}

private fun mapTest() {
    val map = mutableMapOf<Int, String>()
    map.put(1, "MONDAY")
    map.put(2, "TUESDAY")
//    map[1] = "MONDAY"
//    map[2] = "TUESDAY"

    mapOf(1 to "MONDAY", 2 to "TUESDAY")

    for (key in map.keys) {
        println("$key ${map[key]}")
    }

    for ((key, value) in map.entries) {
        println("$key $value")
    }
}

private fun setTest() {
    //    val numbers = setOf(100, 200) // 불변
    val numbers = mutableSetOf(100, 200)    // 가변

    // For Each
    for (number in numbers) {
        println("number = $number")
    }

    // 전통적인 for문
    for ((index, number) in numbers.withIndex()) {
        println("$index $number")
    }
}

private fun listTest() {
    //    val numbers = listOf(100, 200) // 불변 리스트
    val numbers = mutableListOf(100, 200)
    numbers.add(300)

    val emptyList = emptyList<Int>()

    useNumbers(emptyList()) // 타입 추론 가능시, 생략 가능

    // 하나 가져오기
    println(numbers[0])

    // for each
    for (number in numbers) {
        println("number = $number")
    }

    // 전통적인 for문
    for ((index, value) in numbers.withIndex()) {
        println("index, value = $index, $value")
    }
}

private fun useNumbers(numbers: List<Int>) {

}

private fun arrayTest() {
    val arr = arrayOf(100, 200)
    arr.plus(300) // 배열에 새로운 element 추가

    // 방법 1
    for (i in arr.indices) {
        println("$i ${arr[i]}")
    }

    // 방법 2
    for ((idx, value) in arr.withIndex()) {
        println("$idx $value")
    }
}